### PR TITLE
resolver: add EDNS Padding option

### PR DESCRIPTION
**Pull request by xhdix**
close #95 according to RFC8467

I'm not sure it's accurate and you should read rfc8467 again to validate it.

### CODE REVIEW

File: internal/resolver/ooniresolver/ooniresolver.go [hunk 1]
```diff
@@ -101,6 +101,17 @@ func (c *Resolver) newQueryWithQuestion(q dns.Question) (query *dns.Msg) {
...
 	query.Question[0] = q
+	// Add padding for DoH and DoT according to RFC8467
+	if c.Transport().RequiresPadding() {
+		query.SetEdns0(4096, true)
+		// Clients SHOULD pad queries to the closest multiple of 128 octets
+		// rfc8467#section-4.1
+		paddReminder := (128 - query.Len()) % 128
+		qPadding := new([128]byte)
+		qPaddingOpt := new(dns.EDNS0_PADDING)
+		qPaddingOpt.Padding = qPadding[0:paddReminder]
```
Comment (bassosimone) [inline internal/resolver/ooniresolver/ooniresolver.go]: Can this become

```Go
qPaddingOpt.Padding = new([paddReminder]byte)
```

?

Comment (xhdix) [reply internal/resolver/ooniresolver/ooniresolver.go]: No :man_shrugging: 
```
cannot use new([paddReminder]byte) (value of type *[-1]byte) as []byte value in assignment
array length paddReminder (variable of type int) must be constant
```

Comment (bassosimone) [reply internal/resolver/ooniresolver/ooniresolver.go]: Right, I have suggested the wrong change, please try this:

```diff
diff --git a/internal/resolver/ooniresolver/ooniresolver.go b/internal/resolver/ooniresolver/ooniresolver.go
index 8680703..3c4a1d4 100644
--- a/internal/resolver/ooniresolver/ooniresolver.go
+++ b/internal/resolver/ooniresolver/ooniresolver.go
@@ -107,9 +107,8 @@ func (c *Resolver) newQueryWithQuestion(q dns.Question) (query *dns.Msg) {
                // Clients SHOULD pad queries to the closest multiple of 128 octets
                // rfc8467#section-4.1
                paddReminder := (128 - query.Len()) % 128
-               qPadding := new([128]byte)
                qPaddingOpt := new(dns.EDNS0_PADDING)
-               qPaddingOpt.Padding = qPadding[0:paddReminder]
+               qPaddingOpt.Padding = make([]byte, paddReminder)
                query.IsEdns0().Option = append(query.IsEdns0().Option, qPaddingOpt)
        }
        return
```

Comment (bassosimone) [reply internal/resolver/ooniresolver/ooniresolver.go]: (I don't have any other cosmetic comment after this and the next steps are roughly: (1) read the RFC and double check your implementation, (2) write unit tests, (3) merge.)

**Comment by bassosimone** [comment]
Superseded by #150

**Pull request closed without merging**